src/win32/posix_w32.c

```diff
@@ -457,6 +459,10 @@ int p_symlink(const char *target, const char *path)
 		return -1;
 
 	return 0;
+#else
+	/* Symlinks are unavailable before Vista. */
+	return git_futils_fake_symlink(target, path);
```

Comment:
Let's not teach the posix emulation layer about git's notion of fake symlinks.  Can we move this up a level and have whatever "can we do symlinks on this platform" check we have return `false`?

Reply:
I believe this is actually a revert-commit-turned-into-ifdef from before libgit2 was taught about real Win32 symlinks. It's understandable, though I'll have to think of the best way to handle this, since it'll probably be useful for porting to other environments as well.